backend/db/utils.py

```diff
@@ -3,7 +3,7 @@
 todo's
   1. Some functions could be improved using `sql_query()` `params` arg to be less prone to sql injection.
   2. Making 'Connection' optional: Can write a wrapper function and decorate all functions that need, where all it does
-  is `con = con if con else get_db_connection()` and then pass that to inner function.
+  is `conn = con if con else get_db_connection()`, run the inner function, and then close conn if not con.
```

Comment:
## Type 3/3: `conn = con if con else get_db_connection()`
I implemented this pattern basically for convenience to allow us to call SQL functions without having to pass a connection. However, I was not closing these connections.

There's a `todo` here to remind us that it'd be good to have a wrapper for such cases. There's a number of instances of this pattern now.